torch/lib/c10d/CMakeLists.txt

```diff
@@ -81,7 +83,7 @@ if(USE_C10D_GLOO)
   endif()
 endif()
 
-add_library(c10d STATIC ${C10D_SRCS})
+add_library(c10d ${C10D_SRCS})
```

Comment:
@rohan-varma a change made here: c10d is no longer forced to be built as a static lib. I was running into issues where the torchbind static initialization registration code in `CustomClassBindings.cpp` was being discarded, and the canonical "link it with `--whole-archive`" solution produced lots of weird ODR violations in libgcc.

Since I couldn't figure out a reason why this is marked as a static library (as opposed to inheriting the top-level `BUILD_SHARED_LIBS` rule), I opted to remove the `STATIC` property from this.